blender/LilySurfaceScraper/Scrapers/AbstractScraper.py

```diff
@@ -39,16 +41,24 @@ class AbstractScraper():
     scraped_type = {'MATERIAL'}
     # The name of the scraped source, displayed in UI
     source_name = "<Abstract>"
-    # The URL of the source's home, used for the list of availabel sources in panels
+    # The URL of the source's home, used for the list of available sources in panels
     home_url = None
+    # directory of textures
+    home_dir = None
+
+    @staticmethod
+    def sortTextWithNumbers(text):
+        return [int(i) if i.isdigit() else i for i in re.split(r'(\d+)', text)]
 
     @classmethod
     def canHandleUrl(cls, url):
         raise NotImplementedError
 
     def __init__(self, texture_root=""):
+        self.asset_name = None
         self.error = None
         self.texture_root = texture_root
+        self.reinstall = False
 
     @classmethod
     def _fetch(cls, url):
```

Comment:
If `_fetch` still used now that there is `saveFile` and `downloadFunc`?

Reply:
`_fetch` is still being used when an API is being called or just needs a get request, `saveFile` and `downloadFunc` are used to download files.
And `downloadFunc` was a function originally made to remove the duplicate code between `fetchImage` and `fetchZip`.